lib/hls/hls_parser.js

```diff
@@ -1963,10 +2044,58 @@ shaka.hls.HlsParser = class {
           playlist.absoluteUri,
           type);
 
-      references.push(reference);
+      segments.push(reference);
+    }
+
+    // If some segments have sync times, but not all, extrapolate the sync
+    // times of the ones with none.
+    const someSyncTime = segments.some((ref) => ref.syncTime != null);
+    if (someSyncTime) {
+      for (let i = 0; i < segments.length; i++) {
+        const reference = segments[i];
+        if (reference.syncTime != null) {
+          // No need to extrapolate.
+          continue;
+        }
+        // Find the nearest segment with syncTime, in either direction.
+        // This looks forward and backward simultaneously, keeping track of what
+        // to offset the syncTime it finds by as it goes.
+        let forwardAdd = 0;
+        let backwardAdd = 0;
+        for (let j = 0; reference.syncTime == null; j++) {
+          for (const forward of [false, true]) {
+            const other = segments[i + j * (forward ? 1 : -1)];
+            if (other) {
+              if (other.syncTime != null) {
+                reference.syncTime = other.syncTime;
+                reference.syncTime += forward ? forwardAdd : backwardAdd;
+                break;
+              }
+              if (forward) {
+                forwardAdd -= other.endTime - other.startTime;
+              }
+              if (!forward) {
+                backwardAdd += other.endTime - other.startTime;
+              }
+            }
+          }
+        }
+      }
+    } else {
+      // If we want to use program date-time for sync-ing, but there is no such
```

Comment:
As we discussed offline, we should just remove this fallback case altogether. At least until we actually run into a case where it is necessary.